FIREFLY-1536: Support for standalone dispatchTblResultsAdded to support reordering of tables

kpuriIpac opened the pull request:
**Ticket**: https://jira.ipac.caltech.edu/browse/FIREFLY-1536
**IFE-PR**: https://github.com/IPAC-SW/irsa-ife/pull/352

- This PR enables calling `dispatchTblResultsAdded` by itself, indirectly enabling re ordering tables 
    - by removing table from results view by calling `dispatchTblResultsRemove` and then by calling `dispatchTblResultsAdded` when you need to re-add this table in the order you want

**Testing**: 
- Euclid: https://firefly-1536-euclid-reorder-tbls.irsakudev.ipac.caltech.edu/applications/euclid
- Start by doing an Explore Regions search, Inspect Sources search & catalog searches 
- Click on the `Show Results` toggle buttons, if tables are being removed and added back to the results view as expected, then this is working.

loitly reviewed (changes requested):
Sorry, after reviewing this PR, I noticed what I suggested for you to do is not the right approach.  The original implementation is flawed.  It should not have remove `table_ui` state on `TBL_RESULTS_REMOVE`.  It should only remove it from `table_results`.  If you don't mind, I will push a fix to this PR.

kpuriIpac commented:
> I pushed a fix to firefly as well as irsa-ife. Please confirm that it is also working for you. Also, can you add a margin:1 to "Show Results". It is too close to the edge.

done. updated the test build, it all seems to be working. Can you also look at the test build @loitly, and appove [ife PR](https://github.com/IPAC-SW/irsa-ife/pull/352) if all's good there?

loitly commented:
> done. updated the test build, it all seems to be working. Can you also look at the test build @loitly, and appove https://github.com/IPAC-SW/irsa-ife/pull/352 if all's good there?

I already updated test build after I pushed my changes.  My quick test worked.  I did not try to test for anything else.
I thought you only wanted me to look at firefly.  But, I can look at irsa-ife as well.  You should update the config changes, though.

kpuriIpac commented:
> > done. updated the test build, it all seems to be working. Can you also look at the test build @loitly, and appove [IPAC-SW/irsa-ife#352](https://github.com/IPAC-SW/irsa-ife/pull/352) if all's good there?
> 
> I already updated test build after I pushed my changes. My quick test worked. I did not try to test for anything else. I thought you only wanted me to look at firefly. But, I can look at irsa-ife as well. You should update the config changes, though.

Yes it would be nice if you could take a look at ife, but I actually found a bug there anyway that crashes the app when removing tables. So I'll fix that first, before you take a look. I can let you know when that's done. 
 I'm working on config changes in dev brach in parallel since @dencoded said it needs to be pushed ASAP. And the PR may still take some time to be merged (though I think it'll be done today).

kpuriIpac commented:
> > > done. updated the test build, it all seems to be working. Can you also look at the test build @loitly, and appove [IPAC-SW/irsa-ife#352](https://github.com/IPAC-SW/irsa-ife/pull/352) if all's good there?
> > 
> > 
> > I already updated test build after I pushed my changes. My quick test worked. I did not try to test for anything else. I thought you only wanted me to look at firefly. But, I can look at irsa-ife as well. You should update the config changes, though.
> 
> Yes it would be nice if you could take a look at ife, but I actually found a bug there anyway that crashes the app when removing tables. So I'll fix that first, before you take a look. I can let you know when that's done. I'm working on config changes in dev brach in parallel since @dencoded said it needs to be pushed ASAP. And the PR may still take some time to be merged (though I think it'll be done today).

sorry @loitly I got confused doing parallel tasks, I was on dev branch on ife and this feature branch on firefly, hence the bug. there's no actual bug in the PR. so you can test it again if you want, but I think it works.
